File: sherpa-onnx/python/csrc/offline-nemo-enc-dec-ctc-model-config.cc
// sherpa-onnx/python/csrc/offline-nemo-enc-dec-ctc-model-config.cc
//
// Copyright (c)  2023  Xiaomi Corporation

#include "sherpa-onnx/python/csrc/offline-nemo-enc-dec-ctc-model-config.h"

#include <string>
#include <vector>

#include "sherpa-onnx/csrc/offline-nemo-enc-dec-ctc-model-config.h"

namespace sherpa_onnx {

void PybindOfflineNemoEncDecCtcModelConfig(py::module *m) {
  using PyClass = OfflineNemoEncDecCtcModelConfig;
  py::class_<PyClass>(*m, "OfflineNemoEncDecCtcModelConfig")
      .def(py::init<const std::string &>(), py::arg("model"))
      .def_readwrite("model", &PyClass::model)
      .def("__str__", &PyClass::ToString);
}

}  // namespace sherpa_onnx

File: sherpa-onnx/java-api/src/com/k2fsa/sherpa/onnx/OfflineNemoEncDecCtcModelConfig.java
// Copyright 2024 Xiaomi Corporation
package com.k2fsa.sherpa.onnx;

public class OfflineNemoEncDecCtcModelConfig {
    private final String model;

    private OfflineNemoEncDecCtcModelConfig(Builder builder) {
        this.model = builder.model;
    }

    public static Builder builder() {
        return new Builder();
    }

    public String getModel() {
        return model;
    }

    public static class Builder {
        private String model = "";

        public OfflineNemoEncDecCtcModelConfig build() {
            return new OfflineNemoEncDecCtcModelConfig(this);
        }

        public Builder setModel(String model) {
            this.model = model;
            return this;
        }
    }
}
